src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -592,4 +608,15 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		}
 		return notebookCells;
 	}
+
+	private async doChangeKernel(editor: MainThreadNotebookEditor, displayName: string): Promise<boolean> {
+		let listeners = this._modelToDisposeMap.get(editor.id);
+		editor.model.changeKernel(displayName);
```

Comment:
What if model isn't ready yet? Isn't there a case where it's null and we need to reject the promise?